build.gradle

```diff
@@ -83,3 +76,6 @@ processResources
     }
 }
 
+jar {
+    from configurations.shade.collect { it.isDirectory() ? it : zipTree(it) }
+}
```

Comment:
I think to keep the configuration phase fast for gradle I think you want this instead:
```groovy
jar {
     from  { configurations.shade.collect { it.isDirectory() ? it : zipTree(it) } }
    dependsOn configurations.shade
 }
```
Double check with that code to make sure I'm right.